tests/Rules/UsernameTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+use ARKEcosystem\Fortify\Rules\Username;
+
+it('handle null values', function () {
+    $rule = new Username();
+
+    expect($rule->passes('username', null))->toBeFalse();
+});
+
+it('will reject if the value starts with a special character', function () {
```

Comment:
applies to all tests:
- remove the `-`, `!`, etc variants as the checks should only care for `_` and `.` at the moment
- add a new testcase that ensures only numbers, letters and `. _` are allowed while dismissing any other characters (which will catch the `-` and `!` cases)
- add tests that test the regex works correctly, as everything is currently `->toBeFalse()` which we can also achieve if you have a function that simply returns `false`; e.g. it never tests for `true` so we don't know if the rule actually behaves as intended